operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -250,6 +251,19 @@ func CreateOrUpdateLokiStack(
 			ll.Error(err, "failed to check OCP User Workload AlertManager")
 			return err
 		}
+	} else {
+		// Clean up ruler resources
+		err = removeRulesConfigMap(ctx, req, k)
+		if err != nil {
+			ll.Error(err, "failed to remove rules configmap")
+			return err
+		}
+
+		err = removeRuler(ctx, req, k)
+		if err != nil {
+			ll.Error(err, "failed to remove ruler statefulset")
+			return err
+		}
```

Comment:
In addition to the comment below, we can simplify this down to one call remove both or nothing. WDYT?

Reply:
I'd rather keep them separate. Smaller functions with a clear purpose I find is better for many reasons